packages/admin/cms-admin/src/userPermissions/user/UserPage.tsx

```diff
@@ -107,10 +82,30 @@ export const UserPage = ({ userId }: { userId: string }) => {
                     <SupportText>{data.user.email}</SupportText>
                 </ToolbarTitleItem>
                 <ToolbarFillSpace />
-                {isAllowed("impersonation") && (
-                    <CardContent>
-                        <ImpersonationButton userId={userId} />
-                    </CardContent>
+                {/* TODO: remove this condition once more actions are added to that menu */}
+                {userId !== currentUser.id && isAllowed("impersonation") && (
```

Comment:
I would suggest you use there a own component and also use the Early Return pattern for your conditions.